test/quantization/eager/test_numeric_suite_eager.py

```diff
@@ -181,6 +183,54 @@ def compare_and_validate_results(float_model, q_model):
             q_model = quantize_dynamic(model)
             compare_and_validate_results(model, q_model)
 
+    @override_qengines
+    def test_compare_weights_gru_dynamic(self):
+        r"""Compare the weights of float and dynamic quantized LSTM layer"""
+
+        qengine = torch.backends.quantized.engine
+
+        def compare_and_validate_results(float_model, q_model):
```

Comment:
seems that this function has a lot of variations, can we make it reusable and share it between the test cases?